### DISCUSSION TITLE
Future support for Alpine Linux/postmarketOS

### DISCUSSION THREAD

**Discussion by SonicBoone56** [hardinfo2/hardinfo2#87] [Q&A] [answered]
I was wondering if hardinfo2 will eventually be ported to Alpine Linux x86/64 and ARM64, especially since postmarketOS is based on that and would allow a ton of ARM devices to run the benchmarks.

**Comment by hwspeedy** [comment, marked as answer]
Probably yes, as Alpine is also used as desktop on x86/64 and SBC like Raspberry Pi - but it requires an Alpine user to do the work creating an apk package and Alpine Linux to accept the package.

I have added logo+url: https://github.com/hardinfo2/hardinfo2/commit/54bfa6069bbabb3132294675ded416405386dfbf
Add systemV: https://github.com/hardinfo2/hardinfo2/commit/d674c542c2db4684ba12a5726de157cbed289286
Fix CPU Blowfish benchmark for Alpine (musl): https://github.com/hardinfo2/hardinfo2/commit/4deb77938530c9e9e774ecc96df81d99046fa91f + https://github.com/hardinfo2/hardinfo2/commit/18395432ee5e8ea306273df9054c5b9c65d55ae0

**Alpine building: (Tested on x86_64 extended)**
apk add alpine-sdk cmake bash curl
apk add gtk+3.0-dev libsoup3-dev json-glib-dev qt5-qtbase-dev zlib-dev glib-dev
git clone https://github.com/hardinfo2/hardinfo2
cd hardinfo2
DO NOT USE FOR NOW - CHANGES FROM UNSTABLE NEEDED - ./tools/git_latest_release.sh
mkdir build
cd build
cmake ..
make install -j (Will pollute your distro, please package)
apk add lm-sensors sysbench lsscsi mesa-utils dmidecode udisks2 xdg-utils iperf3 fwupd vulkan-tools xrandr
hardinfo2

![image](https://github.com/user-attachments/assets/e03a15d7-e57b-4be9-b24e-8607fc55128e)

**Reply by SonicBoone56** [reply]
How would you get it to run on ARM64? I have a Chromebook I wanted to run hardinfo2 on.

**Reply by hwspeedy** [reply]
The same way - just build with above script.

**Reply by hwspeedy** [reply]
please leave out the ./tools/git_latest_release.sh and thereby get the latest submits for your build - Fixes the service, logo and urls (vendor database).

For packaging remember: -DCMAKE_BUILD_TYPE=Release

**Reply by hwspeedy** [reply]
CPU blowfish benchmark fixed also - please see above for building - updated.

**Comment by SonicBoone56** [comment]
It worked! Unfortunately hardinfo2 seems to not understand what type of CPU the Chromebook has (Mediatek MT8173) or that it even has a GPU in it. Plus it's a laptop, not a SBC. Pretty sure this is the fault of the device or Linux, not hardinfo2. Other than that it worked fine.

![Hardinfo2 Len
[hardinfo2_report_lenovo_100e_mtk_gen2.txt](https://github.com/user-attachments/files/17253404/hardinfo2_report_lenovo_100e_mtk_gen2.txt)
ovo 100e MTK Gen2](https://github.com/user-attachments/assets/8d182e9c-3256-41c9-8d65-724fb5fefbc2)

**Comment by hwspeedy** [comment]
Very good!

The CPU is added by this commit: https://github.com/hardinfo2/hardinfo2/commit/2f3758cf895532d254d9352e2bb1c3880898b039

The GPU does apparently not have drivers in your distro yet - https://www.phoronix.com/news/PowerVR-GX6250-MT8173

The CPU categorizes the system as embedded (SBC) - we need to sort the many machines into groups and that's why it ends up there for now.

llvmpipe is software using a framebuffer (No hardware acceleration).

Have fun,

/hwspeedy

**Comment by hwspeedy** [comment]
added postmarketOS support - distro logo, alpine branch and distro url.

To update your unstable master git and rebuild hardinfo2:
cd hardinfo2/build
git pull
make install -j

**Comment by hwspeedy** [comment]
**Updated Alpine Builld instructions**

Alpine building: (Tested on x86_64 extended)
apk add alpine-sdk cmake bash curl
apk add gtk+3.0-dev libsoup3-dev json-glib-dev qt5-qtbase-dev zlib-dev glib-dev
git clone https://github.com/hardinfo2/hardinfo2
cd hardinfo2
./tools/git_latest_release.sh
mkdir build
cd build
cmake ..
make install -j (Will pollute your distro, please package)
apk add lm-sensors sysbench mesa-utils dmidecode udisks2 xdg-utils iperf3 fwupd vulkan-tools xrandr
hardinfo2

**Comment by hwspeedy** [comment]
**Alpine/PostMarketOS Package Maintainer wanted:**

See - https://gitlab.alpinelinux.org/alpine/aports/-/issues/17052

**Reply by adamthiede** [reply]
Saw your message in irc and made an apkbuild. https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/82241

Could I add you as co-maintainer on it?

**Comment by hwspeedy** [comment]
@adamthiede 

That was fast, thank you so much!

Sure you can add me as co-maintainer - Glad to help - but I know very little of Alpine Linux - just installed it last year to add Alpine Linux support in hardinfo2 for @SonicBoone56.
I can test development and ensure that hardinfo2 releases work on alpine and fix if you spot differences. Both Alpine and PostMarketOS are in the test build server scripts.
Hope you know all the specifics of Alpine/PostMarketOS packaging, thanx.

/hwspeedy

**Reply by hwspeedy** [reply]
@adamthiede 

It looks like the package dependencies are missing?

From README.md: **Package installs these** 
### Depends:
gawk: Used by hardinfo2 service to determine System Type _(Seams to be part of busybox/distrobase)_
**dmidecode: is needed to provide DMI information.
sysbench: ver 1.0.20 - is needed to run standard sysbench benchmarks.
udisks2: is needed to provide storage information.
mesa-utils: glxinfo is needed to get OpenGL info.
lm-sensors: is needed to provide sensors values.
xdg-utils: xdg_open is used to open your browser for bugs, homepage & links.
iperf3: iperf3 is used to benchmark internal network speed.
vulkan-tools: vulkaninfo is used to display vulcan information.
qt5-base: QT5 Framework for QT5 OpenGL GPU Benchmark**

### Recommends/Depends/Optional: (distro choice - prefer installed)
**xrandr/x11-xserver-utils: xrandr is used to read monitor setup**
fwupd: fwupd is used to read and display information about firmware in system.

For Alpine we should probably not install fwupd?
xrandr should be fine as Alpine does not run Wayland only DE.

If dependencies are missing, hardinfo2 is not very usable and users cannot submit good/complete benchmarks for all the wonderful machines, that run Alpine/PostMarketOS.

qt5 framework is the only bigger package - but from SystemRescue, the message was that other programs use it also. Guess the same is true for Alpine/PostMarketOS?

**Reply by adamthiede** [reply]
If a package is listed in `makedepends` and its libraries are used, the library packages are automatically added to its dependencies, so qt5 will be required. fwupd can be optional. Alpine does have wayland only DEs so xrandr shouldn't be a hard requirement. I've updated the apkbuild.

**Reply by adamthiede** [reply]
I've added the rest of the dependencies except for sysbench; it's added only for three architectures that it supports.

**Reply by hwspeedy** [reply]
@adamthiede 

Nice.

sysbench 1.0.20 on riscv64 is not released - but can be build by: 
cd tools; ./build_and_install_sysbench_riscv.sh - fetches luajit from plctlab as luajit is stuck.

Are there other architectures in alpine that does not support luajit?

**Reply by adamthiede** [reply]
I just looked [here](https://pkgs.alpinelinux.org/packages?name=sysbench&branch=edge&repo=&arch=&origin=&flagged=&maintainer=) and selected "all" for architectures and determined sysbench is only on 3 architectures.

**Reply by adamthiede** [reply]
The arches line in this apkbuild https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/sysbench/APKBUILD also says it fails to compile on other arches so I'm fine not including it as a dependency for those other arches. hardinfo2 does compile for everything else (riscv should work as soon as the builders finish rebuilding some dependencies for a library upgrade)

**Comment by hwspeedy** [comment]
Great package view - looks really nice and fast.

luajit is build for many architectures: [Alpine](https://pkgs.alpinelinux.org/packages?name=luajit&branch=edge&repo=&arch=&origin=&flagged=&maintainer=)

sysbench package is limited in Alpine:  [Debian](https://buildd.debian.org/status/package.php?p=sysbench)

**Comment by adamthiede** [comment]
[MR](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/82241) has been merged and hardinfo2 is in Alpine Linux.

**Reply by Saijin-Naib** [reply]
Just started using it, and it works great on both my x86_64 edge desktop/laptop, though the Vulkan tests do not seem to be built/run, despite both systems having vulkan-headers/vulkan-tools installed.

**Reply by hwspeedy** [reply]
@Saijin-Naib

That is because @adamthiede has packaged hardinfo2 version 2.2.10. This was the very first hardinfo2 version, that is packaged in Alpine - so good job. 

Vulkan benchmark is in 2.2.13 and needs packaging changes - see #143

I think @adamthiede is working on version 2.2.13 - see #170 - just a warning - but only shown in version 2.2.13.

**Reply by hwspeedy** [reply]
@Saijin-Naib 

PS: Thanx for testing - please continue the good work, thanx.

**Comment by hwspeedy** [comment]
@adamthiede 

Nice! - good job getting initial 2.2.10 into Alpine based distros.

**You are missing sysbench as dependency - there are 2 choices as I see it:**
1. only build hardinfo2 for architectures supporting both.
2. fixup packaging of sysbench in alpine - sysbench is packages for 10 archs in debian. - https://buildd.debian.org/status/package.php?p=sysbench

**Reply by adamthiede** [reply]
Debian and Alpine aren't the same; Debian's sysbench has quite a few patches, as well as Debian being glibc. I'm also not the maintainer of sysbench in alpine so I'm not really in favor of modifying it.

sysbench doesn't support other architectures apparently:

```
make && \
make install
Detecting operating system.......success [linux]
Detecting machine architecture...failed  [unsupported]
make[1]: *** [Makefile:531: lib/libck.a] Error 1
make[1]: Leaving directory '/builds/adamthiede/aports/community/sysbench/src/sysbench-1.0.20/third_party/concurrency_kit'
make: *** [Makefile:507: all-recursive] Error 1
>>> ERROR: sysbench: build failed
```

You would prefer limiting the architectures of hardinfo2 based on what seems to be an optional dependency? It does run and show a lot of info without sysbench being installed.

**Comment by hwspeedy** [comment]
@adamthiede 

Every half solution comes back to bite us, so we should try to give it a little push.
It should be possible to build sysbench for the same architectures as debian, as sysbench is build with musl for 3 archs in alpine.

Current issue: sysbench build error is ck package - does Alpine build with internal bundled deps - then update ck like below:


```
#update ck for riscv
cd sysbench/third_party/concurrency_kit
rm -rf ck
git clone https://github.com/concurrencykit/ck ck
cd ../../
```

ref: https://github.com/hardinfo2/hardinfo2/blob/master/tools/build_and_install_sysbench_risv.sh

In advance thanx,

/hwspeedy

**Reply by hwspeedy** [reply]
@adamthiede 
PS: The debian patches are just removing internal bundled packages from source (cleanup), use python3 instead of python (from when python2 was active), remove tests from bin (cleanup).

https://sources.debian.org/src/sysbench/1.0.20%2Bds-7/debian/patches/
